app/Services/PlatformUniqueIdService.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace App\Services;
+
+use App\Exceptions\PuidNotUniqueMultipleException;
+use App\Exceptions\PuidNotUniqueSingleException;
+use App\Http\Controllers\Api\v1\StatementAPIController;
+use App\Http\Controllers\Api\v1\StatementMultipleAPIController;
+use App\Models\ArchivedStatement;
+use Exception;
+use Illuminate\Database\QueryException;
+use Illuminate\Http\Response;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\Log;
+use Symfony\Component\Intl\Countries;
+
+class PlatformUniqueIdService
+{
+    protected int $cache_valid_days;
+
+    /**
+     * @param int $cache_valid_days
+     */
+    public function __construct(int $cache_valid_days = 2)
+    {
+        $this->cache_valid_days = $cache_valid_days;
+    }
+
+    public function getCacheKey($platform_id, $puid)
+    {
+        return 'puid-' . $platform_id . '-' . $puid;
+    }
+
+    /**
+     * @param $platform_id
+     * @param $puid
+     * @return void
+     * @throws PuidNotUniqueSingleException
+     */
+    public function addPuidToCache($platform_id, $puid): void
+    {
+        $key = $this->getCacheKey($platform_id, $puid);
+        if (Cache::has($key)) throw new PuidNotUniqueSingleException($puid);
+        Log::info("add to cache: " . $key);
+        Cache::put($key, 0, now()->addDays($this->cache_valid_days));
+    }
+
+    /**
+     * @param $platform_id
+     * @param $puid
+     * @return void
+     * @throws PuidNotUniqueSingleException
+     */
+    public function addPuidToDatabase($platform_id, $puid): void
+    {
+        try {
+            ArchivedStatement::create([
+                'puid' => $puid,
+                'date_received' => Carbon::now(),
+                'platform_id' => $platform_id
+            ]);
+        } catch (QueryException $e) {
+            if (
+                str_contains($e->getMessage(), "platform_puid_unique") || // mysql
+                str_contains($e->getMessage(),
+                    "UNIQUE constraint failed: archived_statements.platform_id, archived_statements.puid") // sqlite
```

Comment:
maybe just a select and use a normal index.

Reply:
good idea. the index is not unique anymore.